cranelift/codegen/src/isa/x86/unwind.rs

```diff
@@ -203,18 +203,30 @@ pub(crate) fn create_unwind_info(
                             let offset = epilogue_pop_offsets[i];
 
                             let reg = func.locations[*arg].unwrap_reg();
-                            unwind_codes.push(UnwindCode::RestoreRegister { offset, reg });
-                            unwind_codes.push(UnwindCode::StackDealloc {
+                            unwind_codes.push((offset, UnwindCode::RestoreRegister { reg }));
+                            unwind_codes.push((
                                 offset,
-                                size: word_size.into(),
-                            });
+                                UnwindCode::StackDealloc {
+                                    size: word_size.into(),
+                                },
+                            ));
+
+                            if Some(reg) == frame_register {
+                                unwind_codes.push((
+                                    offset,
+                                    UnwindCode::SetFramePointer {
```

Comment:
This is really strange to me as this `rsp` isn't really a frame pointer and if this is only necessary to incur a call to `set_cfa_reg` for popping `rbp` in SystemV unwind information generation then I'd prefer it done a different way.

Reply:
Per the discussion in the other comment thread, I think this will be resolved with a new `RestoreFramePointer` unwind code.